.circleci/config.yml

```diff
@@ -183,7 +183,31 @@ commands:
                 fi
               fi
             fi
-          name: "Halt job if code in '<< parameters.dir_names >>' directories are not changed"
+          name: "On PR branch, halt job if code in '<< parameters.dir_names >>' dir are not changed in last commit"
+
+  halt-test-if-e2e:
+    description: "Skip non-e2e test if e2e test files are the only changed files in last commit on PR and master branch"
+    steps:
+      - run:
+          command: |
+            # Always run test on tag builds. Check is done on PR and master branch
+            if [[ -z "$CIRCLE_TAG" ]]; then
+              if [[ "$CIRCLE_BRANCH" == "master" ]]; then
+                # if no changes, exit 0. otherwise, exit 1
+                CHANGES=$(git diff HEAD~1 --quiet --name-only -- . ':!e2e')
```

Comment:
I'd rather just have this run on every master merge. It is good to have this as a source of truth, especially when tests may be flaky. Optimizing on presubmit is fine though